environment.yml

```diff
@@ -0,0 +1,223 @@
+  - libffi=3.2.1
+  - libgcc-ng=9.2.0
+  - libgfortran-ng=7.5.0
+  - libgomp=9.2.0
+  - libiconv=1.15
+  - liblapack=3.8.0
+  - libllvm9=9.0.1
+  - libopenblas=0.3.9
+  - libpng=1.6.37
+  - libsodium=1.0.17
+  - libssh2=1.9.0
+  - libstdcxx-ng=9.2.0
+  - libtiff=4.1.0
+  - libuuid=2.32.1
+  - libwebp-base=1.1.0
+  - libxcb=1.13
+  - libxkbcommon=0.10.0
+  - libxml2=2.9.10
+  - llvm-openmp=10.0.0
+  - lz4-c=1.9.2
+  - make=4.3
+  - markupsafe=1.1.1
+  - matplotlib-base=3.2.2
+  - mistune=0.8.4
+  - mizani=0.7.1
+  - nbconvert=5.6.1
+  - nbformat=5.0.6
+  - ncurses=6.1
+  - notebook=6.0.3
+  - nspr=4.25
+  - nss=3.47
+  - numpy=1.18.5
+  - openssl=1.1.1g
+  - packaging=20.4
+  - palettable=3.3.0
+  - pandas=1.0.5
+  - pandoc=2.9.2.1
+  - pandocfilters=1.4.2
+  - pango=1.42.4
+  - parso=0.7.0
+  - patsy=0.5.1
+  - pcre=8.44
+  - pcre2=10.35
+  - pexpect=4.8.0
+  - pickleshare=0.7.5
+  - pip=20.1.1
+  - pixman=0.38.0
+  - plotnine=0.7.0
+  - prometheus_client=0.8.0
+  - prompt-toolkit=3.0.5
+  - prompt_toolkit=3.0.5
+  - pthread-stubs=0.4
+  - ptyprocess=0.6.0
+  - pycparser=2.20
+  - pygments=2.6.1
+  - pyparsing=2.4.7
+  - pyqt=5.12.3
+  - pyrsistent=0.16.0
+  - python=3.8.3
+  - python-dateutil=2.8.1
+  - python_abi=3.8
+  - pytz=2020.1
+  - pyzmq=19.0.1
+  - qt=5.12.5
+  - qtconsole=4.7.5
+  - qtpy=1.9.0
+  - r=3.6.0
+  - r-assertthat=0.2.1
+  - r-base=3.6.3
+  - r-bit=1.1_14
+  - r-bit64=0.9_7
+  - r-bitops=1.0_6
+  - r-blob=1.1.1
+  - r-boot=1.3_20
+  - r-class=7.3_15
+  - r-cluster=2.0.8
+  - r-codetools=0.2_16
+  - r-dbi=1.0.0
+  - r-digest=0.6.18
+  - r-foreign=0.8_71
+  - r-kernsmooth=2.23_15
+  - r-lattice=0.20_38
+  - r-magrittr=1.5
+  - r-mass=7.3_51.3
+  - r-matrix=1.2_17
+  - r-memoise=1.1.0
+  - r-mgcv=1.8_28
+  - r-nlme=3.1_139
+  - r-nnet=7.3_12
+  - r-pkgconfig=2.0.2
+  - r-plogr=0.2.0
+  - r-prettyunits=1.0.2
+  - r-rcpp=1.0.1
+  - r-rcurl=1.95_4.12
+  - r-recommended=3.6.0
+  - r-rpart=4.1_15
+  - r-rsqlite=2.1.2
+  - r-spatial=7.3_11
+  - r-survival=2.44_1.1
+  - r-xml=3.98_1.19
+  - r-xtable=1.8_4
+  - readline=8.0
+  - rpy2=3.3.2
+  - scikit-learn=0.23.1
+  - scipy=1.4.1
+  - sed=4.7
+  - send2trash=1.5.0
+  - setuptools=47.3.1
+  - simplegeneric=0.8.1
+  - six=1.15.0
+  - sqlite=3.30.1
+  - statsmodels=0.11.1
+  - terminado=0.8.3
+  - testpath=0.4.4
+  - threadpoolctl=2.1.0
+  - tk=8.6.10
+  - tktable=2.10
+  - tornado=6.0.4
+  - tqdm=4.46.1
+  - traitlets=4.3.3
+  - tzlocal=2.1
+  - wcwidth=0.2.5
+  - webencodings=0.5.1
+  - wheel=0.34.2
+  - widgetsnbextension=3.5.1
+  - xorg-kbproto=1.0.7
+  - xorg-libice=1.0.10
+  - xorg-libsm=1.2.3
+  - xorg-libx11=1.6.9
+  - xorg-libxau=1.0.9
+  - xorg-libxdmcp=1.1.3
+  - xorg-libxext=1.3.4
+  - xorg-libxrender=0.9.10
+  - xorg-renderproto=0.11.1
+  - xorg-xextproto=7.3.0
+  - xorg-xproto=7.0.31
+  - xz=5.2.5
+  - zeromq=4.3.2
+  - zipp=3.1.0
+  - zlib=1.2.11
+  - zstd=1.4.4
+  - pip:
+    - pyqt5-sip==4.19.18
+    - pyqtchart==5.12
+    - pyqtwebengine==5.12.1
```

Comment:
Most are dependencies, so don't need to be explicitly added. Since the environment file is used programmatically I prefer to list all the versions in the environment for safety